cmd/honeydipper/integration_test.go

```diff
@@ -109,7 +114,26 @@ func intTestProcesses(t *testing.T) {
 	pidstr, err = exec.CommandContext(ctx, "/usr/bin/pgrep", "-P", ppid).Output()
 	assert.Nil(t, err, "should be able to run pgrep to find all child processes")
 	pids := strings.Split(string(pidstr), "\n")
-	assert.Lenf(t, pids, 11, "expecting 10 child processes for honeydipper process")
+	assert.Lenf(t, pids, 18, "expecting 17 child processes for honeydipper process")
+}
+
+func intTestMakingAPICall(t *testing.T) {
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", "http://localhost:9000/api/events", nil)
+	assert.NoErrorf(t, err, "client error creating http request")
+	req.Header.Add("Authorization", "bearer wrongcredentials")
+	resp, err := client.Do(req)
+	defer resp.Body.Close()
+	assert.NoErrorf(t, err, "client error during api call")
+	assert.Equalf(t, 401, resp.StatusCode, "api call should fail with bad creds")
+
+	req, err = http.NewRequest("GET", "http://localhost:9000/api/events", nil)
+	assert.NoErrorf(t, err, "client error creating http request")
+	req.Header.Add("Authorization", "bearer abcdefg")
+	resp, err = client.Do(req)
+	defer resp.Body.Close()
+	assert.NoErrorf(t, err, "client error during api call")
+	assert.Equalf(t, 200, resp.StatusCode, "api call failed")
```

Comment:
Why is this one a 200 if it’s a failure?

Reply:
The message is displayed only when the test fails, meaning when `resp.StatusCode` is not 200.